### PR TITLE
Split buckets and filesystems URI into endpoints

### PR DESCRIPTION
**Pull request by james-laing**
To enable the exporter to scale and to maintain fast response times for /metrics/array endpoint we need to break off heavy lifting for buckets and filesystems data to their own endpoints.

Proposed fix for https://github.com/PureStorage-OpenConnect/pure-fb-openmetrics-exporter/issues/88.

### CODE REVIEW

**Comment by james-laing** [comment]
This requires discussion as this will require users to target new endpoints for metrics they are currently receiving from /metrics/array.
It will also require all integrations to be updated.

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ GOTEST=$(GOCMD) test
 GOVET=$(GOCMD) vet
 BINARY_NAME=pure-fb-om-exporter
 MODULE_NAME=purestorage/fb-openmetrics-exporter
-VERSION?=1.0.13
+VERSION?=1.0.15-dev
```
Comment (chrroberts-pure) [inline Makefile:6]:
```suggestion
VERSION?=1.0.13
```

File: build/docker/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FROM golang:alpine as build
-ARG VERSION=1.0.13
+ARG VERSION=1.0.15-dev
```
Comment (chrroberts-pure) [inline build/docker/Dockerfile:2]:
```suggestion
ARG VERSION=1.0.13
```

File: cmd/fb-om-exporter/main.go [hunk 3]
```diff
@@ -136,9 +144,15 @@ func main() {
 		http.HandleFunc("/metrics/array", func(w http.ResponseWriter, r *http.Request) {
 			metricsHandler(w, r)
 		})
+		http.HandleFunc("/metrics/buckets", func(w http.ResponseWriter, r *http.Request) {
```
Comment (chrroberts-pure) [inline cmd/fb-om-exporter/main.go:147]: consider renaming to /metrics/objectstore and including purefa_object_store_account metrics as well in the same uri .

**Review by chrroberts-pure — COMMENTED** [review]
i feel wishy washy about this whole thing

**Comment by chrroberts-pure** [comment]
the real 'fix' may be to put in an RFE to increase the number of FS/Buckets we can query at a time .

File: cmd/fb-om-exporter/main.go (hunk 3 above)
Comment (james-laing) [reply cmd/fb-om-exporter/main.go]: Completed in 33abc37dc5707b50af1366c5930ab5313af50bff

File: README.md [hunk 4]
```diff
@@ -128,13 +128,15 @@ Authentication is used by the exporter as the mechanism to cross authenticate to
...
-| http://\<exporter-host\>:\<port\>/metrics          | endpoint       | Full array metrics        |
-| http://\<exporter-host\>:\<port\>/metrics/array    | endpoint       | Array metrics             |
-| http://\<exporter-host\>:\<port\>/metrics/clients  | endpoint       | Clients metrics           |
-| http://\<exporter-host\>:\<port\>/metrics/usage    | endpoint       | Quotas usage metrics      |
-| http://\<exporter-host\>:\<port\>/metrics/policies | endpoint       | NFS policies info metrics |
+| URL                                                   | GET parameters | description               |
+| ------------------------------------------------------| -------------- | --------------------------|
+| http://\<exporter-host\>:\<port\>/metrics             | endpoint       | Full array metrics        |
+| http://\<exporter-host\>:\<port\>/metrics/array       | endpoint       | Array metrics             |
+| http://\<exporter-host\>:\<port\>/metrics/objectstore | endpoint       | Object store metrics      |
```
Comment (chrroberts-pure) [inline README.md:135]:
```suggestion
| http://\<exporter-host\>:\<port\>/metrics/objectstore | endpoint       | Object Store metrics      |
```

**Comment by chrroberts-pure** [comment]
closes #88

**Comment by chrroberts-pure** [comment]
We just need a paragraph for "upgrading to version v1.1.0" in the readme and i think it's good to go

**Comment by james-laing** [comment]
Added detail to new endpoints only being available in v1.1.0

File: specification/metrics/purefb-metrics.md [hunk 5]
```diff
@@ -23,13 +23,16 @@ This document describes the semantic conventions for Pure FlashBlade Metrics.
...
+| Endpoint             | Description              | Metrics Instruments collected                                     |
+| -------------------- | ------------------------ | ----------------------------------------------------------------- |
+| /metrics             | Full array metrics       | all                                                               |
+| /metrics/array       | Array metrics            | `purefb_info`, `purefb_alerts`, `purefb_array`, `purefb_hardware` |
+| /metrics/clients     | Client metrics           | `purefb_info`, `purefb_clients`                                   |
+| /metrics/filesystems | File System metrics      | `purefb_info`, `purefb_file_systems`                              |
+| /metrics/objectstore | Object Store metrics     | `purefb_info`, `purefb_buckets`, `purefb_object_store`            |
+| /metrics/policies    | NFS policy related info  | `purefb_info`, `purefb_nfs_export_rule`                           |
+| /metrics/usage       | Quota metrics            | `purefb_info`, `purefb_file_system_usage`                         |
+
```
Comment (chrroberts-pure) [inline specification/metrics/purefb-metrics.md:35]:
```suggestion
```

**Review by chrroberts-pure — APPROVED** [review]
approved as requested, lgtm.

**Pull request merged**